### PR TITLE
feat: APP-2699 - Use Aragon Gateway for RPC requests

### PR DESCRIPTION
**Pull request by cgero-eth**
## Description

- Update providers handling to use new Aragon Gateway for RPC requests
- Rename `VITE_INFURA_MAINNET_PROJECT_ID` to `VITE_GATEWAY_RPC_API_KEY`

Task: [APP-2699](https://aragonassociation.atlassian.net/browse/APP-2699)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [x]  I have tested my code on the test network.

[APP-2699]: https://aragonassociation.atlassian.net/browse/APP-2699?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/utils/aragonGateway.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  SupportedNetworks,
+  getSupportedNetworkByChainId,
+  infuraApiKey,
+} from './constants';
+import {translateToNetworkishName} from './library';
+
+class AragonGateway {
+  private rpcVersion = '1.0';
+
+  getRpcClient = (
```
Comment (Fabricevladimir) [inline src/utils/aragonGateway.ts]: Why not `getRpcProvider`?

File: src/hooks/useWallet.tsx [hunk 2]
```diff
@@ -62,18 +56,13 @@ export const useWallet = (): IUseWallet => {
...
-        chainId: CHAIN_METADATA[network].id,
-        name: translateToNetworkishName(network),
-        ensAddress:
-          LIVE_CONTRACTS[SupportedVersion.LATEST][
-            translateToNetworkishName(network) as SupportedNetwork
-          ].ensRegistryAddress,
-      });
-    } else return signer?.provider;
-  }, [network, signer?.provider]);
+  const provider = useMemo(
```
Comment (Fabricevladimir) [inline src/hooks/useWallet.tsx:59]: @sepehr2github Do you know why we are bypassing the signer provider on Polygon and Mumbai?

**Review by Fabricevladimir — APPROVED** [review]
Looks good to me, hopefully we caught everything 😅

File: src/utils/aragonGateway.ts (hunk 1 above)
Comment (cgero-eth) [reply src/utils/aragonGateway.ts]: Yeah, it's probably a better name 👌🏻

**Pull request merged**